### PR TITLE
Fix -Wclazy-qstring-arg warnings

### PR DESCRIPTION
**Pull request by Holzhaus**
Avoids unnecessary memory allocations by using the multi-arg overload. See
https://github.com/KDE/clazy/blob/master/docs/checks/README-qstring-arg.md
for details.

### CODE REVIEW

File: src/controllers/hid/hidcontroller.cpp [hunk 1]
```diff
@@ -63,13 +63,12 @@ HidController::HidController(const hid_device_info& deviceInfo, UserSettingsPoin
...
         setDeviceName(
-            QString("%1 %2").arg(hid_product)
-            .arg(hid_serial.right(4)));
+                QString("%1 %2").arg(hid_product, hid_serial.right(4)));
     } else {
-        setDeviceName(
-            QString("%1 %2_%3").arg(hid_product)
-            .arg(hid_serial.right(4))
-            .arg(QString::number(hid_interface_number)));
+        setDeviceName(QString("%1 %2_%3")
```
Comment (uklotzde) [inline src/controllers/hid/hidcontroller.cpp:68]: Will conflict with #3308, but don't mind. Moved into a separate class and replaced by QStringBuilder.

File: src/controllers/midi/midicontroller.cpp [hunk 2]
```diff
@@ -103,10 +103,12 @@ void MidiController::createOutputHandlers() {
...
-                QString("MIDI output message 0x%1 0x%2 has invalid MixxxControl %3, %4")
-                        .arg(QString::number(status, 16).toUpper(),
-                             QString::number(control, 16).toUpper().rightJustified(2,'0'))
-                        .arg(group, key).toUtf8();
+                    QString("MIDI output message 0x%1 0x%2 has invalid MixxxControl %3, %4")
+                            .arg(QString::number(status, 16).toUpper(),
+                                    QString::number(control, 16).toUpper().rightJustified(2, '0'),
+                                    group,
+                                    key)
+                            .toUtf8();
```
Comment (daschuer) [inline src/controllers/midi/midicontroller.cpp:111]: Why toUtf8()? can this be omitted?

**Review by daschuer — APPROVED** [review]
LGTM, waiting for CI

File: src/controllers/midi/midicontroller.cpp (hunk 2 above)
Comment (Holzhaus) [reply src/controllers/midi/midicontroller.cpp:111]: No idea, I just fixed the `.arg()` issues and didn't look into the surrounding code.

**Review by uklotzde — APPROVED** [review]
LGTM

**Pull request merged**